### DISCUSSION TITLE
Single Message Transforms for kafka Connect

### DISCUSSION THREAD

**Discussion by stuartm21** [strimzi/strimzi-kafka-operator#6560] [Q&A]
Hi @scholzj We are planning to install the Single Message Transforms(SMT) along with the aws s3 sink connector. So where we can get the Single Message Transforms jar file or plugins in the internet

**Comment by cstmgl** [comment]
which single message transform? the ones from kafka? or you build your own?

have you tried? https://mvnrepository.com/artifact/org.apache.kafka/connect-transforms

**Reply by stuartm21** [reply]
@cstmgl  We are creating a sink connector in kafka for archiving our data inside the S3 bucket. We want to organize the data like below formate: topics -> customer-1- > 2022 -> 03 -> 24 -> 04:00 (ie; under the S3 bucket we have folder names topics, inside that folder our customer name, customer means to consider each Kafka user as our customer, inside that folder we have year folder, inside that folder we have month folder, inside that folder we have time folder and inside that our data). 

No, I didn't try the above link you provided. Is that help me with the above requirements?

**Reply by cstmgl** [reply]
I think you are talking about the partitioner type not necessary SMT.
But that is more on the usage of the actual connectors not the Strimzi deployment, unless I got it wrong.
I do actually something similar myself with the partitioner that we use a time based partitioner but that really depends on the connector you use.
